test/setup.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+const chai = require('chai');
+
+global.sinon = require('sinon');
+global.assert = chai.assert;
```

Comment:
All these are arguably subjective points, but what is "not a common"? Who defined what is common and what is not? :)
Separating tests' runtime setup from the tests themselves is not a common practice?
Replication of sources fs structure is not common?

What are the downsides? Complication? I don't believe that's the matter.